providers/in-process-flagd/provider.go

```diff
@@ -0,0 +1,644 @@
+package flagd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/open-feature/flagd/core/pkg/eval"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/runtime"
+	"github.com/open-feature/flagd/core/pkg/store"
+	"github.com/open-feature/flagd/core/pkg/sync"
+	"log"
+	sync2 "sync"
+	"time"
+
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+	"os"
+	"strconv"
+)
+
+type ProviderType string
+
+const (
+	defaultMaxSyncRetries      = 5
+	defaultTLS            bool = false
+
+	flagdProxyTLSEnvironmentVariableName          = "FLAGD_PROXY_TLS"
+	flagdProxyCertPathEnvironmentVariableName     = "FLAGD_PROXY_CERT_PATH"
+	flagdMaxSyncRetriesEnvironmentVariableName    = "FLAGD_MAX_SYNC_RETRIES"
```

Comment:
Regarding max retries 🤔 

For flagd remote evaluations, we can define max retries for sync and go offline when limit is reached. This is because we always try to evaluate flags against remote deployment of flagd. So even if sync is shutoff, evaluation can reach flagd when flagd is back online. 

But for in-process evaluations, our flag definitions comes from remote sync service. If this connection breaks, we have to continuously try to connect to the remote flag source. Otherwise, our store will be stale and without a process restart (application restart) we cannot refresh flags. I think this should be added to flagd specification 

cc - @toddbaert & @thisthat

Reply:
That's a valid point @Kavindu-Dodan 👍

Reply:
agree @Kavindu-Dodan , that's a good point - I will adapt the implementation to keep retrying in case of an error

Reply:
if we keep retrying - do you think it still makes sense to eventually go from `STALE` into `ERROR` after a certain amount of retries, or should we get rid of the `MAX_SYNC_RETRIES` env var altogether, and stay in `STALE` for cases where we previously were able to retrieve the flag configs but ran into a interrupted connection afterwards?

Reply:
Good point. For Java provider, I went ahead with status `STALE` and set a max backoff [^1]. If we agree, we can set status to error once we hit the max backoff 🤔 

[^1]: https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java#L29-L30

Reply:
Thank you @Kavindu-Dodan - i think the way you implemented that in the Java library makes the most sense here, so I will apply this logic in this provider as well :)

Reply:
I think if we use the max backoff, it makes sense to make that the "boundary" between STALE and ERROR - but TBH I'm not sure in a server context that anyone would "be okay" with either situation for any substantial amount of time.

I guess a STALE state is tolerable for brief periods and could be characterized as eventual consistency, whereas after max backoff you're at a threshold you've determined to be an error and you might want error-related telemetry to be recorded.

Reply:
Thanks @toddbaert , that makes sense - that's essentially also how the provider is implemented right now. So now this provider will keep retrying with the max backoff being the upper limit, and after exceeding max_retries an error event is sent

Reply:
Yes, it makes sense to emit an error if we reach the max back-off limit. I will add this to Java provider